astropy/units/physical.py

```diff
@@ -1,133 +1,473 @@
+        self._unit = _replace_temperatures_with_kelvin(unit)
+        self._physical_type_id = self._unit._get_physical_type_id()
+        self._physical_type = _standardize_physical_type_names(physical_types)
+        self._physical_type_list = sorted(self._physical_type)
+
+    def __iter__(self):
+        yield from self._physical_type_list
+
+    def __eq__(self, other):
+        """
+        Return `True` if ``other`` represents a physical type that is
+        consistent with the physical type of the `PhysicalType` instance.
+        """
+        if isinstance(other, PhysicalType):
+            return self._physical_type_id == other._physical_type_id
+        elif isinstance(other, str):
+            other = _standardize_physical_type_names(other)
+            return other.issubset(self._physical_type)
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        equality = self.__eq__(other)
+        return not equality if isinstance(equality, bool) else NotImplemented
+
+    @property
+    def _name_string_as_ordered_set(self):
+        return "{" + str(self._physical_type_list)[1:-1] + "}"
+
+    def __repr__(self):
+        if len(self._physical_type) == 1:
+            names = "'" + self._physical_type_list[0] + "'"
+        else:
+            names = self._name_string_as_ordered_set
+        return f"PhysicalType({names})"
+
+    def __str__(self):
+        if len(self._physical_type) == 1:
+            return self._physical_type_list[0]
+        else:
+            return self._name_string_as_ordered_set
+
+    @staticmethod
+    def _dimensionally_compatible_unit(obj):
+        """
+        If a unit is passed in, return that unit.  If a physical type is
+        passed in, return a unit that corresponds to that physical type.
+        If a real number is passed in, return a dimensionless unit.
+        Otherwise, return `NotImplemented`.
+        """
+        if isinstance(obj, core.UnitBase):
+            return _replace_temperatures_with_kelvin(obj)
+        elif isinstance(obj, PhysicalType):
+            return obj._unit
+        elif isinstance(obj, numbers.Real):
+            return core.dimensionless_unscaled
+        else:
+            return NotImplemented
+
+    def _dimensional_analysis(self, other, operation):
+        other_unit = self._dimensionally_compatible_unit(other)
+        if other_unit is NotImplemented:
+            return NotImplemented
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = getattr(self._unit, operation)(other_unit)
+        return new_unit.physical_type
+
+    def __mul__(self, other):
+        return self._dimensional_analysis(other, "__mul__")
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        return self._dimensional_analysis(other, "__truediv__")
+
+    def __rtruediv__(self, other):
+        other = self._dimensionally_compatible_unit(other)
```

Comment:
Couldn't this just be `return self._dimensional_analysis(other, "__rtruediv__")`?

Reply:
No, it turns out that there are a few tests that end up failing when using that one-liner.  I'm curious about why that was happening, but I'm content enough with the current implementation since it's working and still pretty short.

Reply:
I'm still curious what failed... Could you try?

Reply:
I think this is the last older comment I still need to address.  I'll try to get to it tomorrow.

Reply:
@namurphy - did you ever figure out why one cannot use `return self._dimensional_analysis(other, "__rtruediv__")` here?

Reply:
I'm still trying to wrap my head around it, but one thing I noticed is that if we do: 
```Python
>>> u.m.__rtruediv__(1)
<Quantity 1. 1 / m>
```
then we get a `Quantity` which does not have a physical type defined.  In `PhysicalType._dimensional_analysis` the underlying operations are unit operations, and the code is expecting to get the physical type from the resulting unit, and it isn't able to do that from a `Quantity`.  I _think_ that's the reason why we need a workaround.

Reply:
Ah, yes, that's right: one has to do `unit ** -1` to get the inverse unit.

This may be solvable, but does not have to be in this PR! So, I'll resolve.